cmd/btcctl/config.go

```diff
@@ -268,3 +277,59 @@ func loadConfig() (*config, []string, error) {
 
 	return &cfg, remainingArgs, nil
 }
+
+// createDefaultConfig creates a basic config file at the given destination path.
+// For this it tries to read the btcd config file at its default path, and extract
+// the RPC user and password from it.
+func createDefaultConfigFile(destinationPath string) error {
+	// Create the destination directory if it does not exists
+	os.MkdirAll(filepath.Dir(destinationPath), 0700)
+
+	// Read btcd.conf from its default path
+	btcdConfigPath := filepath.Join(btcdHomeDir, "btcd.conf")
+	btcdConfigFile, err := os.Open(btcdConfigPath)
+	if err != nil {
+		return err
+	}
+	defer btcdConfigFile.Close()
+	content, err := ioutil.ReadAll(btcdConfigFile)
+	if err != nil {
+		return err
+	}
+
+	// Extract the rpcuser
+	rpcUserRegexp, err := regexp.Compile(`(?m)^\s*rpcuser=([^\s]+)`)
+	if err != nil {
+		return err
+	}
+	userSubmatches := rpcUserRegexp.FindSubmatch(content)
+	if userSubmatches == nil {
+		// No user found, nothing to do
+		return nil
+	}
+
+	// Extract the rpcpass
+	rpcPassRegexp, err := regexp.Compile(`(?m)^\s*rpcpass=([^\s]+)`)
+	if err != nil {
+		return err
+	}
+	passSubmatches := rpcPassRegexp.FindSubmatch(content)
+	if passSubmatches == nil {
+		// No password found, nothing to do
+		return nil
+	}
+
+	// Create the destination file and write the rpcuser and rpcpass to it
+	dest, err := os.OpenFile(destinationPath, os.O_RDWR|os.O_CREATE|os.O_TRUNC, 0700)
+	if err != nil {
+		return err
+	}
+	defer dest.Close()
+
+	dest.WriteString(fmt.Sprintf(`[Application Options]
```

Comment:
The `[Application Options]` header isn't required, so I would prefer it's left off.  Also, please do this using a regular string with \n so it doesn't mess the source code formatting up.

```Go
	dest.WriteString(fmt.Sprintf("rpcuser=%s\nrpcpass=%s",
		string(userSubmatches[1]), string(passSubmatches[1])))
```